ftp.go

```diff
@@ -503,7 +508,20 @@ func (c *ServerConn) NameList(path string) (entries []string, err error) {
 
 // List issues a LIST FTP command.
 func (c *ServerConn) List(path string) (entries []*Entry, err error) {
-	conn, err := c.cmdDataConnFrom(0, "LIST %s", path)
+	var conn net.Conn
+
+	commands := []string{"MLSD", "LIST"}
+	if _, mlstSupported := c.features["MLST"]; !mlstSupported {
+		commands = commands[1:]
+	}
+
+	for _, cmd := range commands {
+		conn, err = c.cmdDataConnFrom(0, "%s %s", cmd, path)
+		if err == nil {
+			break
+		}
+	}
```

Comment:
I would be even more strict:
```go
var cmd string
var parseFunc func(string)(*Entry, error)

if _, mlstSupported := c.features["MLST"]; mlstSupported {
	cmd = "MLSD"
	parseFunc = parseRFC3659ListLine
} else {
	cmd = "LIST"
	parseFunc = parseListLine
}
```
And then use parseFunc to parse the response.
It someone finds any faulty servers we can still introduce a "support broken servers" flag.